### PR TITLE
cli: add KeyAlreadyExists error

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----
#### Overview
Raise new `KeyAlreadyExists` error when `snapcraft create-key` attempts to create an existing key.

#### Changes
1. Add new error `KeyAlreadyExists`
2. Add unit tests for some other `storeapi` errors.  

#### Source

- https://bugs.launchpad.net/snapcraft/+bug/1979907

(CRAFT-1161)

### CODE REVIEW

File: tests/legacy/unit/store/test_errors.py [hunk 1]
```diff
@@ -113,3 +116,186 @@ def test_snapcraft_exception_handling(
         assert exception.get_details() == expected_details
         assert exception.get_docs_url() == expected_docs_url
         assert exception.get_reportable() == expected_reportable
+
+
+class TestErrorFormatting:
+
+    scenarios = [
```
Comment (mr-cal) [inline tests/legacy/unit/store/test_errors.py:123]: It wasn't necessary to add tests for the other `storeapi` errors, but I see these tests for errors elsewhere in snapcraft, so I decided to add them here too.

**Comment by sergiusens** [comment]
We should re-target this against the release/7.1 branch

**Review by cmatsuoka — COMMENTED** [review]
Fix looks good, could you add a descriptive message to the commit so we can have a good reference of what's happening if we ever need to cherry-pick the commit? Also you can add LP: #1979907 to the commit message to automatically update the launchpad bug.

**Comment by mr-cal** [comment]
Commit message updated

**Review by cmatsuoka — APPROVED** [review]

**Pull request merged**